app/grandchallenge/algorithms/forms.py

```diff
@@ -402,38 +402,46 @@ def __init__(
         structures,
         modalities,
         logo,
+        hide_form,
         *args,
         **kwargs,
     ):
         super().__init__(*args, **kwargs)
-        self.fields["workstation_config"].initial = workstation_config
-        self.fields["workstation_config"].disabled = True
-        self.fields["hanging_protocol"].initial = hanging_protocol
-        self.fields["hanging_protocol"].disabled = True
-        self.fields["view_content"].initial = view_content
-        self.fields["view_content"].disabled = True
-        self.fields["display_editors"].initial = display_editors
-        self.fields["display_editors"].disabled = True
-        self.fields["contact_email"].initial = contact_email
-        self.fields["contact_email"].disabled = True
-        self.fields["workstation"].initial = (
-            workstation
-            if workstation
-            else Workstation.objects.get(
-                slug=settings.DEFAULT_WORKSTATION_SLUG
+        if hide_form:
+            for field in self.fields:
+                self.fields[field].disabled = True
+                self.fields[field].widget = HiddenInput()
```

Comment:
I'm not sure that this works. Wouldn't it be more straightforward to do this client side with a `d-none` class? https://getbootstrap.com/docs/4.0/utilities/display/#hiding-elements

Reply:
We are already not showing the form in the template if `hide_form` is `True`. I initially thought that disabling these fields would add an extra layer of security to make sure people can't just send a post request with form data and get around the limits, but I then realized that for that to work all the other hidden fields need to be set as well, so this can actually be removed.

Reply:
I think it would be more idiomatic to implement the limit check as a form validation method in `form.clean` instead of in the view and passing around this `hide_form` variable.

Reply:
How would that work? We want to either show the form or not, the clean method is only called after a form is submitted, so then we'd always show the form and return validation errors, which is not really what we want, right?

Reply:
Or do you mean keeping the check in the view, but also checking this again in the form's clean method (insetad of passing the `hide_form` variable to the form)?

Reply:
> Or do you mean keeping the check in the view, but also checking this again in the form's clean method (insetad of passing the hide_form variable to the form)?

Yes, you can put the check on the form, call it from the clean method and call it from the view if necessary (also cache it if needed). There may be a better place for the check, maybe on the phase model? Unsure. But yes, raising a validation error on clean is the way to ensure a form cannot be submitted.

Reply:
On the phase model would have been nice, but I couldn't get that to work, so I moved the checks to the form like you suggested